editoast/src/core/v2/pathfinding.rs

```diff
@@ -48,32 +83,9 @@ pub enum PathfindingResult {
         length: u64,
     },
     NotFoundInTracks,
-    IncompatibleElectrification {
-        #[schema(inline)]
-        blocks: Vec<Identifier>,
-        #[schema(inline)]
-        routes: Vec<Identifier>,
-        track_section_ranges: Vec<TrackRange>,
-        length: u64,
-        incompatible_ranges: Vec<(u64, u64)>,
-    },
-    IncompatibleLoadingGauge {
-        #[schema(inline)]
-        blocks: Vec<Identifier>,
-        #[schema(inline)]
-        routes: Vec<Identifier>,
-        track_section_ranges: Vec<TrackRange>,
-        length: u64,
-        incompatible_ranges: Vec<(u64, u64)>,
-    },
-    IncompatibleSignalingSystem {
-        #[schema(inline)]
-        blocks: Vec<Identifier>,
-        #[schema(inline)]
-        routes: Vec<Identifier>,
-        track_section_ranges: Vec<TrackRange>,
-        length: u64,
-        incompatible_ranges: Vec<(u64, u64)>,
+    IncompatibleConstraints {
+        relaxed_constraints_path: Box<PathfindingResultSuccess>,
+        incompatible_constraints: Box<IncompatibleConstraints>,
```

Comment:
It looks possible to avoid Boxing, no?
```suggestion
        relaxed_constraints_path: PathfindingResultSuccess,
        incompatible_constraints: IncompatibleConstraints,
```

Reply:
Nop, I added them yesterday because at the compilation rust tells me to do something ...

Reply:
Ok today I don't issue when I remove the box wrapper, so I removed it

Reply:
It's not at the compilation, it's clippy the linter that says that :
```
error: the `Err`-variant returned from this function is very large
  --> src/views/v2/path/path_item_cache.rs:88:10
   |
88 |       ) -> TrackOffsetResult {
   |            ^^^^^^^^^^^^^^^^^
   |
  ::: src/core/v2/pathfinding.rs:86:5
   |
86 | /     IncompatibleConstraints {
87 | |         relaxed_constraints_path: PathfindingResultSuccess,
88 | |         incompatible_constraints: IncompatibleConstraints,
89 | |     },
   | |_____- the largest variant contains at least 176 bytes
   |
   = help: try reducing the size of `core::v2::pathfinding::PathfindingResult`, for example by boxing large elements or replacing it with `Box<core::v2::pathfinding::PathfindingResult>`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#result_large_err
   = note: `-D clippy::result-large-err` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::result_large_err)]`
  ```
  That's why I added the `Box` wrapper.

Reply:
Not sure what's best here, I will let an editoast maintainer respond (and possibly resolve), here are the ideas I have:
1. Let the Box as proposed
2. Box one level up in the enum (if doable)
3. Box in the 2 Results that trigger clippy (`src/views/v2/path/path_item_cache.rs:88` and `src/views/v2/path/pathfinding.rs:198`)
4. Allow those large Err in results by adding `#[allow(clippy::result_large_err)]`

Reply:
Adding Box is a good option here. There's no reason why this enum should use the IncompatibleConstraints variant more often, so you should reduce its allocation.